### ISSUE TITLE
getIntersectingNodes never detects an intersection when partially=false

### ISSUE THREAD

**Issue by Nick-Lucas** [xyflow/xyflow#3554]
### Describe the Bug

When calculating intersections, getIntersectingNodes tries to calculate the area of the input node with `nodeOrRect.width! * nodeOrRect.height!` but `nodeOrRect` may not have the width and height on it and so may return NaN->false. This may also affect isNodeIntersecting since it uses the same calculation. 

```ts
  const getIntersectingNodes = useCallback<Instance.GetIntersectingNodes<NodeData>>(
    (nodeOrRect, partially = true, nodes) => {
      const [nodeRect, node, isRect] = getNodeRect(nodeOrRect);

      if (!nodeRect) {
        return [];
      }

      return (nodes || store.getState().getNodes()).filter((n) => {
        if (!isRect && (n.id === node!.id || !n.positionAbsolute)) {
          return false;
        }

        const currNodeRect = nodeToRect(n);
        const overlappingArea = getOverlappingArea(currNodeRect, nodeRect);
        const partiallyVisible = partially && overlappingArea > 0;

        return partiallyVisible || overlappingArea >= nodeOrRect.width! * nodeOrRect.height!;
      });
    },
    []
  );
```

The thing is getOverlappingArea successfully does its job, because it's using the result of getNodeRect which is checked and guaranteed to be a rectangle type with x,y,width,height on. So the fix is probably a 1-liner

### Steps to Reproduce the Bug or Issue

Pass a node retrieved from useNodesState to getIntersectingNodes with partially=false and it will never return anything because that node is not a rect and may not have width/height set in all cases.

### Expected behavior

getIntersectingNodes retrieves the rect if necessary (it already does) and uses it for all calculations (it doesn't do this)

### Platform

- OS: [macOS]
- Browser: [Chrome]
- Version: [11.9.4]

**Comment by moklick** [comment]
I'll close it, when the new version is published @Nick-Lucas

**Comment by Nick-Lucas** [comment]
Apologies! I assumed I'd mis-tagged my PR and you hadn't noticed 😅

Thanks for merging!

**Comment by moklick** [comment]
Thanks for your help here. Fixed in 11.10.1
